File: dashlane-project/Dashlane/src/main/java/com/dashlane/dagger/singleton/JsonModule.java
package com.dashlane.dagger.singleton;

import com.dashlane.util.GsonJsonSerialization;
import com.dashlane.util.JsonSerialization;
import com.google.gson.Gson;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;



@Module
public class JsonModule {

    @Provides
    @Singleton
    public JsonSerialization provideJsonSerialization() {
        return new GsonJsonSerialization(new Gson());
    }
}

File: dashlane-project/Dashlane/src/main/java/com/dashlane/dagger/singleton/JsonModule.kt
package com.dashlane.dagger.singleton

import com.dashlane.util.GsonJsonSerialization
import com.dashlane.util.JsonSerialization
import com.google.gson.Gson
import dagger.Module
import dagger.Provides
import dagger.hilt.InstallIn
import dagger.hilt.components.SingletonComponent
import javax.inject.Singleton

@Module
@InstallIn(SingletonComponent::class)
object JsonModule {
    @Provides
    @Singleton
    fun provideJsonSerialization(): JsonSerialization = GsonJsonSerialization(Gson())
}
